staging/src/k8s.io/kubectl/pkg/metricsutil/metrics_printer_test.go

```diff
@@ -0,0 +1,374 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metricsutil
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	metricsapi "k8s.io/metrics/pkg/apis/metrics"
+)
+
+func TestPrintNodeMetrics(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		nodeMetric     []metricsapi.NodeMetrics
+		nodes          []*v1.Node
+		noHeader       bool
+		sortBy         string
+		expectedErr    error
+		expectedOutput string
+	}{
+		{
+			name:  "Single node with default header",
+			nodes: []*v1.Node{newNode("node1")},
+			nodeMetric: []metricsapi.NodeMetrics{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node1", ResourceVersion: "10", Labels: map[string]string{"key": "value"}},
+					Window:     metav1.Duration{Duration: time.Minute},
+					Usage: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("1"),
+						v1.ResourceMemory: resource.MustParse("1Gi"),
+					},
+				},
+			},
+			expectedOutput: "NAME    CPU(cores)   CPU(%)   MEMORY(bytes)   MEMORY(%)   \nnode1   1000m        10%      1024Mi          10%         \n",
```

Comment:
All the `expectedOutput`s can use raw string instead of this?

Reply:
sure!  It would be much clearer using the raw string output.
done